src/libraries/System.Text.Json/tests/JsonStringTests.cs

```diff
@@ -110,6 +110,23 @@ public static void TestDateTimeOffset(string testStr, string expectedStr)
             Assert.Equal(dateTimeOffset, dateTimeOffset2);
         }
 
+        [Theory]
+        [MemberData(nameof(JsonDateTimeTestData.ValidISO8601Tests), MemberType = typeof(JsonDateTimeTestData))]
+        public static void TestDateTimeAndDateTimeOffsetWithEscapedChars(string testStr, string iso8601Str)
+        {
+            var jsonString = new JsonString(testStr);
```

Comment:
I would trim the `testStr` and remove the embedded quotes.

Keep in mind, the strings in `ValidISO8601Tests` are meant for entire JSON documents. That's why they contain the quotes.

`JsonString` points to a single token within a JSON document, so it wouldn't contain the embedded quotes.

For example:
```C#
string json = "[\"1997-07-16T19:20:30.0000001\", \"hi\", 5]";

JsonArray node = (JsonArray)JsonNode.Parse(json);

Console.WriteLine(((JsonString)node[0]).GetDateTime()); // 7/16/1997 7:20:30 PM
Console.WriteLine(((JsonString)node[1])); // hi
Console.WriteLine(((JsonNumber)node[2]).GetInt32()); // 5

json = "[\"\"1997-07-16T19:20:30.0000001\"\", \"hi\", 5]"; // This is invalid JSON (unescaped embedded quotes not allowed in strings)

json = "[\"\\\"1997-07-16T19:20:30.0000001\\\"\", \"hi\", 5]"; // The string is no longer a DateTime and hence GetDateTime() should throw
```